test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -0,0 +1,357 @@
+                    directory,
+                    relativePath);
+
+                // TODO: get rid of the fallback options when we have a consistent convention
+
+                if (!File.Exists(_profilerFileLocation))
+                {
+                    _output.WriteLine($"Attempt 1: Unable to find profiler at {_profilerFileLocation}.");
+                    // Let's try the executing directory, as dotnet publish ignores the Copy attributes we currently use
+                    _profilerFileLocation = Path.Combine(
+                        GetExecutingProjectBin(),
+                        relativePath);
+                }
+
+                if (!File.Exists(_profilerFileLocation))
+                {
+                    _output.WriteLine($"Attempt 2: Unable to find profiler at {_profilerFileLocation}.");
+                    // One last attempt at the actual native project directory
+                    _profilerFileLocation = Path.Combine(
+                        GetProfilerProjectBin(),
+                        fileName);
+                }
+
+                if (!File.Exists(_profilerFileLocation))
+                {
+                    throw new Exception($"Attempt 3: Unable to find profiler at {_profilerFileLocation}");
+                }
+
+                _output.WriteLine($"Found profiler at {_profilerFileLocation}.");
+            }
+
+            return _profilerFileLocation;
+        }
+
+        public string GetSampleApplicationPath()
+        {
+            var appFileName = EnvironmentHelper.IsCoreClr()
+                                  ? $"Samples.{SampleName}.dll"
+                                  : $"Samples.{SampleName}.exe";
+            var sampleAppPath = Path.Combine(GetSampleApplicationOutputDirectory(), appFileName);
+            return sampleAppPath;
+        }
+
+        public string GetSampleApplicationOutputDirectory()
+        {
+            var solutionDirectory = GetSolutionDirectory();
+            var sampleDirectory = $"Samples.{SampleName}";
+
+            var binDir = Path.Combine(
+                solutionDirectory,
+                _samplesDirectory,
+                sampleDirectory,
+                "bin");
+
+            string outputDir;
+
+            if (EnvironmentHelper.GetOS() == "win")
+            {
+                outputDir = Path.Combine(
+                    binDir,
+                    GetPlatform(),
+                    GetBuildConfiguration(),
+                    GetTargetFramework(),
+                    EnvironmentHelper.GetRuntimeIdentifier());
+            }
+            else
+            {
+                outputDir = Path.Combine(
+                    binDir,
+                    GetBuildConfiguration(),
+                    GetTargetFramework(),
+                    EnvironmentHelper.GetRuntimeIdentifier(),
+                    "publish");
+            }
+
+            return outputDir;
+        }
+
+        public string GetTargetFramework()
```

Comment:
Part of my goal in this class is to reduce dependency on compile flags.
Then we can decorate tests with something along the lines of: 
`[IgnoreOnFramework]`
`[IgnoreOnFramework("4.5")]`
`[IgnoreOnCore]`
`[IgnoreOnCore("2")]`

Reply:
I would probably go with a single attribute that takes the same target framework monikers that msbuild uses. Something like:
```csharp
[IgnoreOnTargetFramework("net45")]
[IgnoreOnTargetFramework("netcoreapp2.0")]
```

Reply:
the reason I'd opt for straight version numbers is for when we want to do something like.... 
"Only run on .net framework 4.6 and up"